components/viaduct/src/backend.rs

```diff
@@ -49,3 +50,56 @@ pub fn send(request: crate::Request) -> Result<crate::Response, crate::Error> {
         self::ffi::send(request)
     }
 }
+
+pub fn validate_request(request: &crate::Request) -> Result<(), crate::Error> {
+    if request.url.scheme() != "https"
+        && request.url.host_str() != Some("localhost")
+        && request.url.host_str() != Some("127.0.0.1")
+    {
+        return Err(crate::Error::NonTlsUrl);
+    }
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::validate_request;
+    #[test]
+    fn test_validate_request() {
+        let _https_request = crate::Request::new(
+            crate::Method::Get,
+            url::Url::parse("https://www.facebook.com").unwrap(),
```

Comment:
Tiny nit: I think we should use a "neutral" domain here e.g. `example.com` rather than `facebook.com`, as a matter of general test hygiene. (I'm sure it won't make any difference in this case, just a general probably-shouldn't-scatter-references-to-other-companies-products-throughout-our-code-if-we-dont-have-to kind of vibe...)